From 1ec303966905fa0f7947db3d18a61223f19563b8 Mon Sep 17 00:00:00 2001
From: W-Mark Kubacki <wmark@hurrikane.de>
Date: Mon, 30 Dec 2013 17:53:46 +0100
Subject: [PATCH] 3072 bit dhparam

obtained using
  quantum-random-numgen 52M -out random-data-for-gen.bin
  openssl dhparam -rand random-data-for-gen.bin -C 3072
---
 src/event/ngx_event_openssl.c | 64 +++++++++++++++++++++++++++++++------------
 1 file changed, 46 insertions(+), 18 deletions(-)

diff --git a/src/event/ngx_event_openssl.c b/src/event/ngx_event_openssl.c
index ee66713..4ac70ad 100644
--- a/src/event/ngx_event_openssl.c
+++ b/src/event/ngx_event_openssl.c
@@ -599,27 +599,55 @@ ngx_ssl_dhparam(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *file)
 
     /*
      * -----BEGIN DH PARAMETERS-----
-     * MIGHAoGBALu8LcrYRnSQfEP89YDpz9vZWKP1aLQtSwju1OsPs1BMbAMCducQgAxc
-     * y7qokiYUxb7spWWl/fHSh6K8BJvmd4Bg6RqSp1fjBI9osHb302zI8pul34HcLKcl
-     * 7OZicMyaUDXYzs7vnqAnSmOrHlj6/UmI0PZdFGdX2gcd8EXP4WubAgEC
+     * MIIBiAKCAYEAiwABDneWkUjSFr4z6IDLVWqtC/HG8jahsAxzYoC4tG8pUYOiAUJv
+     * x/GnIoUj6uBbRxeDIQPB4taLk438Y9bIRIyJPcZuH7v4upYbQJNaVjjMNH47I8/c
+     * 2Ky6y/hti20E8DQOoW3aUKRsyZVtNP/BySGfAxYPXUh7JVfaNCImBnUnqcXZQwhZ
+     * +sZPa7T1+WnmUiw59DF/ZOjY/JjI/wNUfA1hygCKIElnhf+a4fwP5T1k/gWfhxhu
+     * am1TA/2WNtGccP0Y5Z3yhWRumXzdx1xHgDGSs4DvEAiPDncUSVYSWdNryezS4wrd
+     * t5dbdZOgiX3Z0kr2gYnaJwLnxUKz6ARg0RX2yChUAp0Bm4aNKPJgsIbwCcrnUyey
+     * K0QSvPaEDIguhXp6LarHevSEAoRzeRUlYUXkgkaTVoGLrzLjz9bxJNCKj10+nvTt
+     * RXqaf0nu/QR1yN1bKSIp6YJYmhmKmdBAIxPtyjxUYbbkbRED9Bzh6n562tWRGYAq
+     * j55K0AZ5BTTbAgEC
      * -----END DH PARAMETERS-----
      */
 
-    static unsigned char dh1024_p[] = {
-        0xBB, 0xBC, 0x2D, 0xCA, 0xD8, 0x46, 0x74, 0x90, 0x7C, 0x43, 0xFC, 0xF5,
-        0x80, 0xE9, 0xCF, 0xDB, 0xD9, 0x58, 0xA3, 0xF5, 0x68, 0xB4, 0x2D, 0x4B,
-        0x08, 0xEE, 0xD4, 0xEB, 0x0F, 0xB3, 0x50, 0x4C, 0x6C, 0x03, 0x02, 0x76,
-        0xE7, 0x10, 0x80, 0x0C, 0x5C, 0xCB, 0xBA, 0xA8, 0x92, 0x26, 0x14, 0xC5,
-        0xBE, 0xEC, 0xA5, 0x65, 0xA5, 0xFD, 0xF1, 0xD2, 0x87, 0xA2, 0xBC, 0x04,
-        0x9B, 0xE6, 0x77, 0x80, 0x60, 0xE9, 0x1A, 0x92, 0xA7, 0x57, 0xE3, 0x04,
-        0x8F, 0x68, 0xB0, 0x76, 0xF7, 0xD3, 0x6C, 0xC8, 0xF2, 0x9B, 0xA5, 0xDF,
-        0x81, 0xDC, 0x2C, 0xA7, 0x25, 0xEC, 0xE6, 0x62, 0x70, 0xCC, 0x9A, 0x50,
-        0x35, 0xD8, 0xCE, 0xCE, 0xEF, 0x9E, 0xA0, 0x27, 0x4A, 0x63, 0xAB, 0x1E,
-        0x58, 0xFA, 0xFD, 0x49, 0x88, 0xD0, 0xF6, 0x5D, 0x14, 0x67, 0x57, 0xDA,
-        0x07, 0x1D, 0xF0, 0x45, 0xCF, 0xE1, 0x6B, 0x9B
+    static unsigned char dh3072_p[] = {
+        0x8B,0x00,0x01,0x0E,0x77,0x96,0x91,0x48,0xD2,0x16,0xBE,0x33,
+        0xE8,0x80,0xCB,0x55,0x6A,0xAD,0x0B,0xF1,0xC6,0xF2,0x36,0xA1,
+        0xB0,0x0C,0x73,0x62,0x80,0xB8,0xB4,0x6F,0x29,0x51,0x83,0xA2,
+        0x01,0x42,0x6F,0xC7,0xF1,0xA7,0x22,0x85,0x23,0xEA,0xE0,0x5B,
+        0x47,0x17,0x83,0x21,0x03,0xC1,0xE2,0xD6,0x8B,0x93,0x8D,0xFC,
+        0x63,0xD6,0xC8,0x44,0x8C,0x89,0x3D,0xC6,0x6E,0x1F,0xBB,0xF8,
+        0xBA,0x96,0x1B,0x40,0x93,0x5A,0x56,0x38,0xCC,0x34,0x7E,0x3B,
+        0x23,0xCF,0xDC,0xD8,0xAC,0xBA,0xCB,0xF8,0x6D,0x8B,0x6D,0x04,
+        0xF0,0x34,0x0E,0xA1,0x6D,0xDA,0x50,0xA4,0x6C,0xC9,0x95,0x6D,
+        0x34,0xFF,0xC1,0xC9,0x21,0x9F,0x03,0x16,0x0F,0x5D,0x48,0x7B,
+        0x25,0x57,0xDA,0x34,0x22,0x26,0x06,0x75,0x27,0xA9,0xC5,0xD9,
+        0x43,0x08,0x59,0xFA,0xC6,0x4F,0x6B,0xB4,0xF5,0xF9,0x69,0xE6,
+        0x52,0x2C,0x39,0xF4,0x31,0x7F,0x64,0xE8,0xD8,0xFC,0x98,0xC8,
+        0xFF,0x03,0x54,0x7C,0x0D,0x61,0xCA,0x00,0x8A,0x20,0x49,0x67,
+        0x85,0xFF,0x9A,0xE1,0xFC,0x0F,0xE5,0x3D,0x64,0xFE,0x05,0x9F,
+        0x87,0x18,0x6E,0x6A,0x6D,0x53,0x03,0xFD,0x96,0x36,0xD1,0x9C,
+        0x70,0xFD,0x18,0xE5,0x9D,0xF2,0x85,0x64,0x6E,0x99,0x7C,0xDD,
+        0xC7,0x5C,0x47,0x80,0x31,0x92,0xB3,0x80,0xEF,0x10,0x08,0x8F,
+        0x0E,0x77,0x14,0x49,0x56,0x12,0x59,0xD3,0x6B,0xC9,0xEC,0xD2,
+        0xE3,0x0A,0xDD,0xB7,0x97,0x5B,0x75,0x93,0xA0,0x89,0x7D,0xD9,
+        0xD2,0x4A,0xF6,0x81,0x89,0xDA,0x27,0x02,0xE7,0xC5,0x42,0xB3,
+        0xE8,0x04,0x60,0xD1,0x15,0xF6,0xC8,0x28,0x54,0x02,0x9D,0x01,
+        0x9B,0x86,0x8D,0x28,0xF2,0x60,0xB0,0x86,0xF0,0x09,0xCA,0xE7,
+        0x53,0x27,0xB2,0x2B,0x44,0x12,0xBC,0xF6,0x84,0x0C,0x88,0x2E,
+        0x85,0x7A,0x7A,0x2D,0xAA,0xC7,0x7A,0xF4,0x84,0x02,0x84,0x73,
+        0x79,0x15,0x25,0x61,0x45,0xE4,0x82,0x46,0x93,0x56,0x81,0x8B,
+        0xAF,0x32,0xE3,0xCF,0xD6,0xF1,0x24,0xD0,0x8A,0x8F,0x5D,0x3E,
+        0x9E,0xF4,0xED,0x45,0x7A,0x9A,0x7F,0x49,0xEE,0xFD,0x04,0x75,
+        0xC8,0xDD,0x5B,0x29,0x22,0x29,0xE9,0x82,0x58,0x9A,0x19,0x8A,
+        0x99,0xD0,0x40,0x23,0x13,0xED,0xCA,0x3C,0x54,0x61,0xB6,0xE4,
+        0x6D,0x11,0x03,0xF4,0x1C,0xE1,0xEA,0x7E,0x7A,0xDA,0xD5,0x91,
+        0x19,0x80,0x2A,0x8F,0x9E,0x4A,0xD0,0x06,0x79,0x05,0x34,0xDB,
     };
 
-    static unsigned char dh1024_g[] = { 0x02 };
+
+    static unsigned char dh3072_g[] = { 0x02 };
 
 
     if (file->len == 0) {
@@ -630,8 +658,8 @@ ngx_ssl_dhparam(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *file)
             return NGX_ERROR;
         }
 
-        dh->p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), NULL);
-        dh->g = BN_bin2bn(dh1024_g, sizeof(dh1024_g), NULL);
+        dh->p = BN_bin2bn(dh3072_p, sizeof(dh3072_p), NULL);
+        dh->g = BN_bin2bn(dh3072_g, sizeof(dh3072_g), NULL);
 
         if (dh->p == NULL || dh->g == NULL) {
             ngx_ssl_error(NGX_LOG_EMERG, ssl->log, 0, "BN_bin2bn() failed");
-- 
1.8.5.2

